lib/colorado_lottery.rb

```diff
@@ -0,0 +1,50 @@
+
+
+class ColoradoLottery
+  attr_reader :registered_contestants, :winners, :current_contestants
+
+  def initialize
+    @registered_contestants = {}
+    @winners = []
+    @current_contestants = {}
+  end
+
+  def find_game_interest(name, game_interest)
+    name.game_interests.find do |game|
+        if game_interest.name == game
+         return true
+       else
+         return false
+        end
+    end
+  end
+
+  def interested_and_18?(name, game_interest)
+    if name.age >= 18 || find_game_interest(name, game_interest) == true
+      true
+    else
+      false
+    end
+  end
+
+  def can_register?(name, game)
```

Comment:
`can_register` should be a bit more complex - it should check if the contestant is interested, over 18, and either a CO resident, or the game is national.